### PR TITLE
Linux package

### PR DESCRIPTION
**Pull request by w00lf**

### CODE REVIEW

**Comment by ronaldtse** [comment]
@w00lf instead of changing the workflow, let’s add a new workflow for Linux vs macOS?

**Comment by ronaldtse** [comment]
Could we do Ruby 2.6 since it’s supposed to be faster? Thanks.

**Comment by w00lf** [comment]
> Could we do Ruby 2.6 since it’s supposed to be faster? Thanks.

Ok, i have changed it

> @w00lf instead of changing the workflow, let’s add a new workflow for Linux vs macOS?

I have created a separate workflow for linux

**Comment by ronaldtse** [comment]
@repu1sion (Andrii) this is the Linux PR, which was confirmed by @w00lf to fully work except for the memory usage of 22GB, which exceeds GHA machines of 7GB with 14GB SSD.

Please help reduce the memory usage (probably at https://github.com/metanorma/ruby-packer/issues/4) to make this job run. 

Thanks!

**Comment by ronaldtse** [comment]
@w00lf , @repu1sion has an issue with compiling on Linux:

```sh
configure: error: cannot find install-sh, install.sh, or shtool in build-aux "."/build-aux
Failed running [{"CI"=>"true", "ENCLOSE_IO_USE_ORIGINAL_RUBY"=>"1", "CFLAGS"=>" -fPIC -O3 -fno-fast-math -ggdb3 -Os -fdata-sections -ffunction-sections -pipe ", "LDFLAGS"=>""}, "./configure", "--with-pic", "--enable-libgdbm-compat", "--disable-shared", "--enable-static", "--without-readline", "--prefix=/tmp/rubyc/local"]
Makefile:6: recipe for target 'compile' failed
make: *** [compile] Error 1
```

Could you please let him know? Thanks!

**Comment by ronaldtse** [comment]
@w00lf : @repu1sion says that building with `clang` on Linux works and doesn't run into the memory problem. 

From @repu1sion:

> It seems like there is an issue with gcc which is yet not fixed:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=12245
> 12245 – [8/9/10 regression] Uses lots of memory when compiling large initialized arrays
https://gcc.gnu.org

> ruby-packer generates 400Mb *.c file with 100Mb array in it

file looks like:

```c
#include <stdint.h>
#include <stddef.h>

const uint8_t enclose_io_memfs[114798592] = { 104
,115,113,115,104,55,0,0,25,234,210,93,0,0,2,0,239,1,0,0,1,0,17,0,192,0,1,0,4,0,0,0,48,8,227,225,1,0,0,0,27,167,215,6,0,0,0,0,19,167,215,6,0,0,0,0,255,255,255,255,255,255,255,255,107,96,211,6,0,0,0,0,71,69,213,6,0,0,0,0,200,84,215,6,0,0,0,0,157,166,215,6,0,0,0,0,120,218,140,189,7,92
...
};
```

> so whole file is just binary array which uses later to create squashfs image etc
> during compilation process gcc have to read all array in memory
> and also creates metadata for every array element like for a tree node
> so we have 100M of array elements, like 150 bytes for each ~ 15.2Gb of RAM needed

> On my machine with 8Gb RAM + 8GB swap it works

> so bigger swapfile or switch to another compiler, clang for example

**Comment by ronaldtse** [comment]
> also it could be possible to rework ruby-packer to not create such huge array

I think this needs to be done one day. But maybe compiler change to `clang` is easiest for now. 

We will need a new issue to make ruby-packer use a file on disk instead of a static memory huge array...

**Comment by w00lf** [comment]
> > also it could be possible to rework ruby-packer to not create such huge array
> 
> I think this needs to be done one day. But maybe compiler change to `clang` is easiest for now.
> 
> We will need a new issue to make ruby-packer use a file on disk instead of a static memory huge array...

Got it, will try to use clang and if it will not work will add swapfile

**Comment by ronaldtse** [comment]
@w00lf any luck so far? Eager to get things working 👍

**Comment by w00lf** [comment]
> @w00lf any luck so far? Eager to get things working 👍

ping @ronaldtse. Can you review this pr?

File: bin/build.sh [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 [ -d build ] || mkdir build; rm -rf build/* || true
 
+# TODO: wrong test for linux
```
Comment (ronaldtse) [inline bin/build.sh]: Do we have to fix this test?

**Comment by ronaldtse** [comment]
Other than the question above, the rest are good. Feel free to merge it! Thanks @w00lf!

File: bin/build.sh (hunk 1 above)
Comment (ronaldtse) [reply bin/build.sh]: @w00lf I'll merge first, see if this needs to be fixed later.

**Pull request merged**